File: VECTOAux/VectoAuxiliaries/Hvac/ISSMGenInputs.vb

Namespace Hvac

    Public Interface ISSMGenInputs

        'Bus Parameterisation	
        Property BP_BusModel As String
        Property BP_NumberOfPassengers As Double
        Property BP_BusFloorType As String
        Property BP_DoubleDecker As Boolean
        Property BP_BusLength As Double
        Property BP_BusWidth As Double
        Property BP_BusHeight As Double

        ReadOnly Property BP_BusFloorSurfaceArea As Double
        ReadOnly Property BP_BusWindowSurface As Double
        ReadOnly Property BP_BusSurfaceAreaM2 As Double
        ReadOnly Property BP_BusVolume As Double

        'Boundary Conditions:			
        Property BC_GFactor As Double
        ReadOnly Property BC_SolarClouding As Double
        ReadOnly Property BC_HeatPerPassengerIntoCabinW As Double
        Property BC_PassengerBoundaryTemperature As Double
        ReadOnly Property BC_PassengerDensityLowFloor As Double
        ReadOnly Property BC_PassengerDensitySemiLowFloor As Double
        ReadOnly Property BC_PassengerDensityRaisedFloor As Double
        ReadOnly Property BC_CalculatedPassengerNumber As Double
        ReadOnly Property BC_UValues As Double
        Property BC_HeatingBoundaryTemperature As Double
        Property BC_CoolingBoundaryTemperature As Double
        ReadOnly Property BC_TemperatureCoolingTurnsOff As Double
        Property BC_HighVentilation As Double
        Property BC_lowVentilation As Double
        ReadOnly Property BC_High As Double
        ReadOnly Property BC_Low As Double
        ReadOnly Property BC_HighVentPowerW As Double
        ReadOnly Property BC_LowVentPowerW As Double
        Property BC_SpecificVentilationPower As Double
        Property BC_AuxHeaterEfficiency As Double
        Property BC_GCVDieselOrHeatingOil As Double
        ReadOnly Property BC_WindowAreaPerUnitBusLength As Double
        ReadOnly Property BC_FrontRearWindowArea As Double
        Property BC_MaxTemperatureDeltaForLowFloorBusses As Double
        Property BC_MaxPossibleBenefitFromTechnologyList As Double

        'EnviromentalConditions				
        Property EC_EnviromentalTemperature As Double
        Property EC_Solar As Double
        ReadOnly Property EC_EnvironmentalConditionsMap As IEnvironmentalConditionsMap
        Property EC_EnviromentalConditions_BatchFile As String
        Property EC_EnviromentalConditions_BatchEnabled As Boolean

        'AC-system				            
        Property AC_CompressorType As String
        ReadOnly Property AC_CompressorTypeDerived As String
        Property AC_CompressorCapacitykW As Double
        ReadOnly Property AC_COP As Double

        'Ventilation				
        Property VEN_VentilationOnDuringHeating As Boolean
        Property VEN_VentilationWhenBothHeatingAndACInactive As Boolean
        Property VEN_VentilationDuringAC As Boolean
        Property VEN_VentilationFlowSettingWhenHeatingAndACInactive As String
        Property VEN_VentilationDuringHeating As String
        Property VEN_VentilationDuringCooling As String

        'Aux. Heater				
        Property AH_EngineWasteHeatkW As Double
        Property AH_FuelFiredHeaterkW As Double

    End Interface

End Namespace

File: VECTOAux/BusAuxiliaries/Interfaces/DownstreamModules/HVAC/ISSMGenInputs.cs
namespace DownstreamModules.HVAC
{
	public interface ISSMGenInputs
	{

		// Bus Parameterisation	
		string BP_BusModel { get; set; }
		double BP_NumberOfPassengers { get; set; }
		string BP_BusFloorType { get; set; }
		bool BP_DoubleDecker { get; set; }
		double BP_BusLength { get; set; }
		double BP_BusWidth { get; set; }
		double BP_BusHeight { get; set; }

		double BP_BusFloorSurfaceArea { get; }
		double BP_BusWindowSurface { get; }
		double BP_BusSurfaceAreaM2 { get; }
		double BP_BusVolume { get; }

		// Boundary Conditions:			
		double BC_GFactor { get; set; }
		double BC_SolarClouding { get; }
		double BC_HeatPerPassengerIntoCabinW { get; }
		double BC_PassengerBoundaryTemperature { get; set; }
		double BC_PassengerDensityLowFloor { get; }
		double BC_PassengerDensitySemiLowFloor { get; }
		double BC_PassengerDensityRaisedFloor { get; }
		double BC_CalculatedPassengerNumber { get; }
		double BC_UValues { get; }
		double BC_HeatingBoundaryTemperature { get; set; }
		double BC_CoolingBoundaryTemperature { get; set; }
		double BC_TemperatureCoolingTurnsOff { get; }
		double BC_HighVentilation { get; set; }
		double BC_lowVentilation { get; set; }
		double BC_High { get; }
		double BC_Low { get; }
		double BC_HighVentPowerW { get; }
		double BC_LowVentPowerW { get; }
		double BC_SpecificVentilationPower { get; set; }
		double BC_AuxHeaterEfficiency { get; set; }
		double BC_GCVDieselOrHeatingOil { get; set; }
		double BC_WindowAreaPerUnitBusLength { get; }
		double BC_FrontRearWindowArea { get; }
		double BC_MaxTemperatureDeltaForLowFloorBusses { get; set; }
		double BC_MaxPossibleBenefitFromTechnologyList { get; set; }

		// EnviromentalConditions				
		double EC_EnviromentalTemperature { get; set; }
		double EC_Solar { get; set; }
		IEnvironmentalConditionsMap EC_EnvironmentalConditionsMap { get; }
		string EC_EnviromentalConditions_BatchFile { get; set; }
		bool EC_EnviromentalConditions_BatchEnabled { get; set; }

		// AC-system				            
		string AC_CompressorType { get; set; }
		string AC_CompressorTypeDerived { get; }
		double AC_CompressorCapacitykW { get; set; }
		double AC_COP { get; }

		// Ventilation				
		bool VEN_VentilationOnDuringHeating { get; set; }
		bool VEN_VentilationWhenBothHeatingAndACInactive { get; set; }
		bool VEN_VentilationDuringAC { get; set; }
		string VEN_VentilationFlowSettingWhenHeatingAndACInactive { get; set; }
		string VEN_VentilationDuringHeating { get; set; }
		string VEN_VentilationDuringCooling { get; set; }

		// Aux. Heater				
		double AH_EngineWasteHeatkW { get; set; }
		double AH_FuelFiredHeaterkW { get; set; }
		double AH_FuelEnergyToHeatToCoolant { get; set; }
		double AH_CoolantHeatTransferredToAirCabinHeater { get; set; }
	}
}
